lib/rubocop/cop/rspec/metadata_style.rb

```diff
@@ -0,0 +1,219 @@
+        def_node_matcher :match_ambiguous_trailing_metadata?, <<~PATTERN
+          (send _ _ _ ... !{hash sym})
+        PATTERN
+
+        def on_metadata(symbols, hash)
+          symbols.each do |symbol|
+            on_metadata_symbol(symbol)
+          end
+
+          return unless hash
+
+          on_empty_metadata_hash(hash) if hash.pairs.empty?
+
+          hash.pairs.each do |pair|
+            on_metadata_pair(pair)
+          end
+        end
+
+        private
+
+        def autocorrect_pair(corrector, node)
+          remove_pair(corrector, node)
+          insert_symbol(corrector, node)
+        end
+
+        def autocorrect_symbol(corrector, node)
+          return if match_ambiguous_trailing_metadata?(node.parent)
+
+          remove_symbol(corrector, node)
+          insert_pair(corrector, node)
+        end
+
+        def bad_metadata_pair?(node)
+          style == :symbol && match_boolean_metadata_pair?(node)
+        end
+
+        def bad_metadata_symbol?(_node)
+          style == :hash
+        end
+
+        def format_symbol_to_pair_source(node)
+          "#{node.value}: true"
+        end
+
+        def insert_pair(corrector, node)
+          hash_node = extract_metadata_hash(node.parent)
+          if hash_node.nil?
+            insert_pair_as_last_argument(corrector, node)
+          elsif hash_node.pairs.any?
+            insert_pair_to_non_empty_hash_metadata(corrector, node, hash_node)
+          else
+            insert_pair_to_empty_hash_metadata(corrector, node, hash_node)
+          end
+        end
+
+        def insert_pair_as_last_argument(corrector, node)
+          corrector.insert_before(
+            node.parent.location.end || node.parent.source_range.with(
+              begin_pos: node.parent.source_range.end_pos
+            ),
+            ", #{format_symbol_to_pair_source(node)}"
+          )
+        end
+
+        def insert_pair_to_empty_hash_metadata(corrector, node, hash_node)
+          corrector.insert_after(
+            hash_node.location.begin,
+            " #{format_symbol_to_pair_source(node)} "
+          )
+        end
+
+        def insert_pair_to_non_empty_hash_metadata(corrector, node, hash_node)
+          corrector.insert_after(
+            hash_node.children.last,
+            ", #{format_symbol_to_pair_source(node)}"
+          )
+        end
+
+        def insert_symbol(corrector, node)
+          corrector.insert_after(
+            node.parent.left_sibling,
+            ", #{node.key.value.inspect}"
+          )
+        end
+
+        def message_for_style
+          format(
+            'Use %<style>s style for metadata.',
+            style: style
+          )
+        end
+
+        def on_empty_metadata_hash(node)
+          return if style == :hash
+
+          add_offense(node, message: 'Remove empty metadata.') do |corrector|
+            remove_empty_metadata(corrector, node)
+          end
+        end
```

Comment:
ask) I guess we could cut out this to another cop with a name like `RSpec/EmptyMetadata`. What do you think?
If we can keep it small, I want to keep it small, because the cost of maintenance goes up a bit when one cop work increases.

Reply:
I'm fine either way. @rubocop/rubocop-rspec WDYT ?

Reply:
Sounds good to have two cops. Easier than splitting later.

Reply:
Agree on splitting now.

Reply:
Okay, so I'm going to split it into 2 cops 👌